packages/babel-plugin-transform-block-scoping/src/validation.ts

```diff
@@ -177,7 +177,9 @@ function injectTDZChecks(binding: Scope.Binding, state: PluginPass) {
       }
     } else if (path.isAssignmentExpression()) {
       const nodes = [];
-      const ids = path.getBindingIdentifiers();
+      const ids = process.env.BABEL_8_BREAKING
+        ? path.getAssignmentIdentifiers()
+        : path.getBindingIdentifiers();
```

Comment:
This change is deferred to Babel 8 because plugins might be used with an old Babel core that depends on a legacy Babel traverse.